build_tools/github/test_source.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -e
+set -x
+
+python -m venv test_env
+source test_env/bin/activate
+
+python -m pip install dist/*.tar.gz
+python -m pip install pytest pandas
+
+# Run the tests on the installed source distribution
+mkdir tmp_for_test
+cd tmp_for_test
+
+pytest --pyargs sklearn
```

Comment:
This is weird, the failure suggests that scikit-learn has been installed in dev mode:

```
+ mkdir tmp_for_test
+ cd tmp_for_test
+ pytest --pyargs sklearn
ImportError while loading conftest '/home/runner/work/scikit-learn/scikit-learn/conftest.py'.
../conftest.py:15: in <module>
    from sklearn.utils import _IS_32BIT
../sklearn/__init__.py:81: in <module>
    from . import __check_build  # noqa: F401
../sklearn/__check_build/__init__.py:46: in <module>
    raise_build_error(e)
../sklearn/__check_build/__init__.py:31: in raise_build_error
    raise ImportError("""%s
E   ImportError: No module named 'sklearn.__check_build._check_build'
```

Reply:
Or maybe this is because of a `PYTHONPATH` variable that would put `${github.workspace} == "/home/runner/work/scikit-learn/scikit-learn/"` first when importing `sklearn`?